array/array.mbt

```diff
@@ -81,6 +81,45 @@ test "iteri" {
   }
 }
 
+pub fn iter_rev[T](self : Array[T], f : (T) -> Unit) {
+  let mut i = self.length() - 1
+  while i >= 0 {
+    f(self[i])
+    i = i - 1
+  }
```

Comment:
we can write it in a function style:
```moonbit
for i =  self.length() = 1; i >= 0 ; i = i - 1{
   f(self(i)
}
```

Reply:
I found that the `map_with_index` in the array library uses Functional loop, so I think I should apply this feature to modifiable positions in both the array and string libraries.

Reply:
I have already adopted functional loops for both array and string, and the name of this pull request will be changed to be more accurate.